src/main/java/com/hazelcast/web/WebFilter.java

```diff
@@ -512,8 +517,9 @@ private String findHazelcastSessionIdFromRequest() {
                     }
                 }
             }
-            // if hazelcast session id is not found on the cookie, look into request parameters
-            if (hzSessionId == null) {
+            // if hazelcast session id is not found on the cookie and using request parameter is enabled, look into
+            // request parameters
+            if (hzSessionId == null && useRequestParameter) {
```

Comment:
Having a sensible default seems more important than backwards compatibility in this particular case as the default behaviour of the filter causes issues now and this functionality is a rather less frequently used one. We're choosing to make it `false` by default and let people enable it if they need to use request parameter for hazelcast session id.